File: Python/nstress_coreduction_cycle.py
#!/usr/bin/env python3
# -*- encoding: utf-8 -*-


# output several bash script

from re import A
import sys
import pandas as pd 

list = ["#!/bin/bash\n# The interpreter used to execute the script\n\n",\
            "# This project is to explore the phase diagram (Ru=11) with\n", \
            "# different rigidity ratio, halfwidth and characteristic slip distance \n\n",\
            "##SBATCH directives that convey submission options:\n\n",\
            "##SBATCH --mail-user=zhai5108@gmail.com\n",\
            "##SBATCH --mail-type=BEGIN,END\n",\
            "#SBATCH --nodes=1\n",\
            "#SBATCH --mem=30000m\n",\
            "#SBATCH --time=14-00:00:00\n",\
            "#SBATCH --partition=standard\n\n"]

# for high resolution, the mem should be large enough!!  default value is 20G
# but for highest resolution 20: the mem should be about 60G


par = pd.read_table("./key_par_nc.txt", header = 0, sep=" ")
print(par.columns)
print(len(par))
ratio = par['ratio']
halfwidth = par['halfwidth']
L = par['L']
multiple = par['multiple']
cos_reduction = par['cos_reduction']


n = 20              # label of tasks
N = 8              # multi threads for each task: 4 is good!!

with open('nstress_coreduction_cycle.sh','w') as f:       
    f.writelines(list)
    f.write("#SBATCH --account=yiheh0\n")
    f.write("#SBATCH --ntasks-per-node=%.0f \n\n" %(N))           # request N cores
    f.write("# The application(s) to execute along with its input arguments and options:\n")
    f.write("# half-width(m) rigidity_ratio Lc(m) multiple(normal stress) cos_reduction\n\n")
    f.write("#Uncomment the following words to submit different simulations!!! \n\n\n")
    for i in range(0,len(par)):              # here from task 2
            n = n+1
            f.write("##SBATCH --job-name=case%.0f \n"  %(n))
            f.write("##SBATCH --output=/home/%u/log/%x-%j.log \n")
            f.write("##SBATCH --error=/home/%u/log/error-%x-%j.log \n")
            f.write("#julia --threads %.0f run.jl %.2f %.0f %.6f %.0f %.2f \n\n"\
                %(N,ratio.iloc[i],halfwidth.iloc[i],L.iloc[i], multiple.iloc[i], cos_reduction.iloc[i]))

File: Matlab/nstress_coreduction_cycle.m
%%
clear
close all
gamma = pi/4;  % empirical constant parameter 
r = 0.85;   % the shear wave reduction=20%  1-0.2=0.8  r is the rigidity ratio
mu = 3.203812032e10;  % Pa 
H = 1500;      % m  half-width of damage zone   
%sigma = 50e6;  % Pa
a = 0.015;
b = 0.019;
L = 0.008;
multiple = [4,5,6,7];    % Pa
cos_reduction = [0.05,0.06,0.07,0.08];
W = 10000;
fid=fopen('key_par_nc.txt','w');   % rigidity ratio, half width, Lc, cohesive zone size
fprintf(fid,'ratio halfwidth L multiple cos_reduction\n');
for i = 1:length(multiple)
    for j = 1:length(cos_reduction)
         A = [r, H ,L ,multiple(i),cos_reduction(j)];
        fprintf(fid,'%.6f %.6f %.6f %.6f %.6f\n',A);
    end
end
fclose(fid);







